test/jdk/java/awt/Mouse/MouseDragEnterExitTest.java

```diff
@@ -0,0 +1,187 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Frame;
+import java.awt.Panel;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.MouseMotionListener;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 4141361
+ * @summary Test to Ensures that mouse enter / exit is delivered to a new
+ *          frame or component during a drag
+ * @library /open/test/jdk/java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual MouseDragEnterExitTest
+ */
+
+public class MouseDragEnterExitTest {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                Click on the blue frame, drag to the white frame, and back
+                You should get enter/exit messages for the frames when dragging
+                """;
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(35)
+                .testUI(MouseEvents.initialize())
+                .logArea(8)
+                .build()
+                .awaitAndCheck();
+    }
+}
+
+class MouseEvents extends Frame implements MouseListener,
+        MouseMotionListener {
+    static int WITH_WIDGET = 0;
+
+    public MouseEvents(int mode) {
+        super("Mouse Drag Enter/Exit Test");
+        setSize(300, 300);
+
+        addMouseListener(this);
+        addMouseMotionListener(this);
+
+        if (mode == WITH_WIDGET) {
+            setLayout(new BorderLayout());
+            add("Center", new SimplePanel());
+        }
+    }
+
+    public void mouseMoved(MouseEvent e) {
+    }
+
+    public void mouseDragged(MouseEvent e) {
+    }
+
+    public void mouseClicked(MouseEvent e) {
+    }
+
+    public void mouseEntered(MouseEvent e) {
+        printEvent(e);
+    }
+
+
+    public void mouseExited(MouseEvent e) {
+        printEvent(e);
+    }
+
+    public void mousePressed(MouseEvent e) {
+    }
+
+    public void mouseReleased(MouseEvent e) {
+    }
+
+    public static void printEvent(MouseEvent e) {
+        Component source = e.getComponent();
+
+        String type = "none";
+
+        if (e.getID() == MouseEvent.MOUSE_PRESSED)
+            type = "MOUSE_PRESSED";
+
+        if (e.getID() == MouseEvent.MOUSE_CLICKED)
+            type = "MOUSE_CLICKED";
+
+        if (e.getID() == MouseEvent.MOUSE_RELEASED)
+            type = "MOUSE_RELEASED";
+
+        if (e.getID() == MouseEvent.MOUSE_MOVED)
+            type = "MOUSE_MOVED";
+
+        if (e.getID() == MouseEvent.MOUSE_ENTERED)
+            type = "MOUSE_ENTERED";
+
+        if (e.getID() == MouseEvent.MOUSE_ENTERED)
+            type = "MOUSE_ENTERED";
+
+        if (e.getID() == MouseEvent.MOUSE_EXITED)
+            type = "MOUSE_EXITED";
+
```

Comment:
MOUSE_ENTERED is used twice...Also, it seems you are only printing MouseExit/Enter so other types can be deleted..

Reply:
Updated.